File: app/src/main/java/com/example/mg/masterdetail/UI/WeatherDayRecyclerViewAdapter.java
package com.example.mg.masterdetail.UI;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.RequestManager;
import com.example.mg.masterdetail.R;
import com.example.mg.masterdetail.UI.DI.IFragScope;
import com.example.mg.masterdetail.data.model.Weather10daysModel.ForecastBean.SimpleforecastBean.ForecastdayBeanX;

import java.util.List;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;

@IFragScope
public class WeatherDayRecyclerViewAdapter
        extends RecyclerView.Adapter<WeatherDayRecyclerViewAdapter.ViewHolder> {

    private List<ForecastdayBeanX> mValues;
    private RequestManager mGlide;

    @Inject
    public WeatherDayRecyclerViewAdapter(RequestManager glide) {
        mGlide = glide;
    }

    public void setData(List<ForecastdayBeanX> values) {
        mValues = values;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view =
                LayoutInflater.from(parent.getContext())
                        .inflate(R.layout.day_list_content,
                                parent,
                                false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        holder.weatherDay.setText(String.format("%s %s",
                mValues.get(position).getDate().getWeekday_short(),
                mValues.get(position).getDate().getDay()
        ));
        mGlide
                .load(mValues.get(position).getIcon_url())
                .into(holder.weatherIc);
        holder.weatherDegree.setText(String.format("%s / %s",
                mValues.get(position).getLow().getCelsius(),
                mValues.get(position).getHigh().getCelsius()));

    }

    @Override
    public int getItemCount() {
        try {
            return mValues.size();
        } catch (NullPointerException e) {
            return 0;
        }
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.text_weather_day)
        TextView weatherDay;
        @BindView(R.id.image_weather)
        ImageView weatherIc;
        @BindView(R.id.text_weather_degree)
        TextView weatherDegree;


        ViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/java/com/example/mg/masterdetail/ui/WeatherDayRecyclerViewAdapter.kt
package com.example.mg.masterdetail.ui

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.RequestManager
import com.example.mg.masterdetail.R
import com.example.mg.masterdetail.data.model.WeatherDaysModel
import kotlinx.android.synthetic.main.day_list_content.view.*

class WeatherDayRecyclerViewAdapter(
        private val mGlide: RequestManager,
        private var mValues: List<WeatherDaysModel.Data>
) : RecyclerView.Adapter<WeatherDayRecyclerViewAdapter.ViewHolder>() {

    fun setData(values: List<WeatherDaysModel.Data>) {
        mValues = values
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.day_list_content,
                        parent,
                        false)
        return ViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.itemView.weatherDay.text = mValues[position].valid_date

        holder.itemView.weatherDegree.text = "${mValues[position].app_max_temp} / ${mValues[position].low_temp}"

        mValues[position].weather?.let {
            mGlide.load("https://www.weatherbit.io/static/img/icons/${it.icon}.png")
                    .into(holder.itemView.weatherIc)
        }

    }

    override fun getItemCount(): Int = mValues.size

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view)

}